conference/src/main/java/com/kyriba/conference/management/service/ScheduleServiceImpl.java

```diff
@@ -0,0 +1,120 @@
+package com.kyriba.conference.management.service;
+
+import com.kyriba.conference.management.api.dto.PresentationRequest;
+import com.kyriba.conference.management.api.dto.TopicDto;
+import com.kyriba.conference.management.dao.HallRepository;
+import com.kyriba.conference.management.dao.PresentationRepository;
+import com.kyriba.conference.management.domain.Hall;
+import com.kyriba.conference.management.domain.Presentation;
+import com.kyriba.conference.management.domain.Topic;
+import com.kyriba.conference.management.domain.exception.EntityNotFound;
+import com.kyriba.conference.management.domain.exception.LinkedEntityNotFound;
+import com.kyriba.conference.management.domain.exception.TimeslotIsBooked;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+
+@Service
+public class ScheduleServiceImpl implements ScheduleService
+{
+  private final PresentationRepository presentationRepository;
+
+  private final HallRepository hallRepository;
+
+
+  @Autowired
+  public ScheduleServiceImpl(PresentationRepository presentationRepository,
+                             HallRepository hallRepository)
+  {
+    this.presentationRepository = presentationRepository;
+    this.hallRepository = hallRepository;
+  }
+
+
+  @Override
+  public Iterable<Presentation> getSchedule()
+  {
+    return presentationRepository.findAll();
+  }
+
+
+  @Transactional
+  @Override
+  public Long addPresentation(PresentationRequest presentationRequest) throws LinkedEntityNotFound
+  {
+    Optional<Hall> hall = hallRepository.findById(presentationRequest.getHall());
+    if (hall.isPresent()) {
```

Comment:
You could use chain methods on Optional instead of if/else

Reply:
done